Structure_Engine/Modify/SetElements0D.cs

```diff
@@ -49,9 +50,25 @@ public static Bar SetElements0D(this Bar bar, List<IElement0D> newElements0D)
                 return null;
             }
 
-            Bar clone = bar.GetShallowClone() as Bar;
-            clone.StartNode = newElements0D[0] as Node;
-            clone.EndNode = newElements0D[1] as Node;
+            Bar clone = bar.DeepClone() as Bar;
+
+            // Default the Bars end if the input is an Point
+            if (newElements0D[0] is Point)
+            {
+                clone.StartNode = Create.Node(newElements0D[0] as Point);
+                clone.Release.StartRelease = Create.FixConstraint6DOF();
```

Comment:
Another thing I realised. We will want to null check the release class here before assigning to it. Basically only set the start release here if the Release != null

Reply:
oh, good catch